homeassistant/components/foscam/camera.py

```diff
@@ -111,6 +190,37 @@ def disable_motion_detection(self):
             _LOGGER.debug("Communication problem")
             self._motion_status = False
 
+    async def async_perform_ptz(self, movement, travel_time):
+        """Perform a PTZ action on the camera."""
+        _LOGGER.debug("PTZ action '%s' on %s", movement, self._name)
+
+        ret, _ = await self.hass.async_add_executor_job(
+            {
+                DIR_UP: self._foscam_session.ptz_move_up,
+                DIR_DOWN: self._foscam_session.ptz_move_down,
+                DIR_LEFT: self._foscam_session.ptz_move_left,
+                DIR_RIGHT: self._foscam_session.ptz_move_right,
+                DIR_TOPLEFT: self._foscam_session.ptz_move_top_left,
+                DIR_TOPRIGHT: self._foscam_session.ptz_move_top_right,
+                DIR_BOTTOMLEFT: self._foscam_session.ptz_move_bottom_left,
+                DIR_BOTTOMRIGHT: self._foscam_session.ptz_move_bottom_right,
+            }[movement]
```

Comment:
Please separate the dict lookup from the call on different lines to make it more readable.